### PR TITLE
my modest contribution, dnn in real time

### PR DESCRIPTION
**Pull request by cabelo**
### This pullrequest changes

My small and modest contribution, dnn in real time, see video bellow..

https://www.youtube.com/watch?v=NHtRlndE2cg

### CODE REVIEW

File: samples/dnn/yolo_object_cam.cpp [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+			const int probability_index = 5;
+			const int probability_size = detectionMat.cols - probability_index;
+			float *prob_array_ptr = &detectionMat.at<float>(i, probability_index);
+
+			size_t objectClass = std::max_element(prob_array_ptr, prob_array_ptr + probability_size) - prob_array_ptr;
+			float confidence = detectionMat.at<float>(i, (int)objectClass + probability_index);
+
+			if (confidence > confidenceThreshold)
+			{
+					string sLabel = returnLabel(objectClass,mylist );
```
Comment (alalek) [inline samples/dnn/yolo_object_cam.cpp]: Please replace tabs to 4 spaces.

File: samples/dnn/yolo_object_cam.cpp [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+					cv::Mat color(roi.size(), CV_8UC3, cv::Scalar(0, 0, 0)); 
+					double alpha = 0.3;
+					cv::addWeighted(color, alpha, roi, 1.0 - alpha , 0.0, roi); 
+		 
+					putText(frame, sLabel, Point(object.x, object.y+12),FONT_HERSHEY_PLAIN, 1 ,Scalar::all(255)); 
+			
+			}
+		}
+		 
+		imshow("GPU", frame); //waitKey();
```
Comment (alalek) [inline samples/dnn/yolo_object_cam.cpp]: "GPU" -> "Result" / "YOLO"
(by default GPU acceleration is not enabled)

File: samples/dnn/yolo_object_cam.cpp [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+		if (cv::waitKey(30) >= 0) grabFrame = false;
```
Comment (alalek) [inline samples/dnn/yolo_object_cam.cpp]:
```.diff
-if (cv::waitKey(30) >= 0) grabFrame = false;
+if (cv::waitKey(30) > 0)
+    break;
```

File: samples/dnn/yolo_object_cam.cpp [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+const char* params
+    = "{ help           | false | print usage         }"
+      "{ cam            |       | device index        }"
+      "{ cfg            |       | model configuration }"
+      "{ model          |       | model weights       }"
+      "{ labels         |       | label of the object }"
+      "{ min_confidence | 0.24  | min confidence      }";
+
+
+void assetRoi(Rect &_roi, Mat &_frame  )
```
Comment (alalek) [inline samples/dnn/yolo_object_cam.cpp]: Add "static" for local functions "assetRoi" / "addLabels" / "returnLabel"

**Comment by vpisarev** [comment]
@cabelo, thanks for the contribution!

what's the reason of populating the same code with little changes? Can we instead modify yolo_object_detection.cpp and add extra features there?

File: samples/dnn/yolo_object_cam.cpp (hunk 4 above)
Comment (cabelo) [reply samples/dnn/yolo_object_cam.cpp]: Done

File: samples/dnn/yolo_object_cam.cpp (hunk 3 above)
Comment (cabelo) [reply samples/dnn/yolo_object_cam.cpp]: done

File: samples/dnn/yolo_object_cam.cpp (hunk 2 above)
Comment (cabelo) [reply samples/dnn/yolo_object_cam.cpp]: Done

File: samples/dnn/yolo_object_cam.cpp (hunk 1 above)
Comment (cabelo) [reply samples/dnn/yolo_object_cam.cpp]: Done

**Comment by cabelo** [comment]
screenshot_20171129_211955
Add image example.

**Comment by alalek** [comment]
> perf_video-ippicv-opencl

This is sporadic failure, you need to amend commit and make a force push (or just wait when OpenCV maintainers reschedule these failed builds).

PNG image is too large. Lossless format is not necessary for samples (it is not a strict test), so try to use JPEG format instead.

Please merge/replace your changes into [yolo_object_detection.cpp](https://github.com/opencv/opencv/blob/3.3.1/samples/dnn/yolo_object_detection.cpp) file (we should avoid samples with similar/extended functionality).

**Comment by vpisarev** [comment]
@cabelo, let me close this PR. Open the new one where you modify the existing yolo_object_detection.cpp. Also, we would rather not to add any new images with people into opencv repository, because it may violate legal rules. Let's only leave the live mode of the demo where the data is captured from camera (or let the user specify video file or still image). Thanks for understanding.

**Comment by cabelo** [comment]
@vpisarev  if i crop my face in picture?

**Pull request closed without merging**